src/main/java/hudson/plugins/jabber/im/transport/JabberIMConnection.java

```diff
@@ -312,13 +311,25 @@ private boolean createConnection() throws XMPPException, SaslException,
 		
 		String serviceName = desc.getServiceName();
 		final XMPPTCPConnectionConfiguration.Builder cfg = XMPPTCPConnectionConfiguration.builder();
+		
+		if (pi.getProxyType() == ProxyType.NONE) {
+			cfg.setSocketFactory(SocketFactory.getDefault());
+		} else {
+			cfg.setProxyInfo(pi);
+		}
+		
 		if (serviceName == null) {
-			cfg.setHost(hostnameOverride).setPort(port).setProxyInfo(pi);
+			cfg
+			    .setHost(hostnameOverride)
+			    .setPort(port);
 		} else if (this.hostnameOverride == null) {
-		    // uses DNS lookup, to get the actual hostname for this service: 
-			cfg.setServiceName(serviceName).setProxyInfo(pi);
+		    // uses DNS lookup, to get the actual hostname for this service:			
+			cfg.setServiceName(serviceName);
 		} else {
-			cfg.setHost(hostnameOverride).setPort(port).setServiceName(serviceName).setProxyInfo(pi);
+			cfg
```

Comment:
That may very well be true. The point here is that you mix different concerns (bumping smack to 4.1.9 **and** improving the readability) into the same branch, which makes it harder for the plugin maintainer to review and digest the resulting "Bump Smack to 4.1.9" commit, which we will send upstream, which makes it harder to get the change accepted.